grpc-examples/README.md

```diff
@@ -0,0 +1,236 @@
+}
+```
+
+It will allow you to operate on a single row. When executing the above snippet, it will return:
+```yaml
+[string: "a"
+, int: 1
+]
+```
+ 
+### Batch Query
+
+In case we want to execute N queries, we can use the `executeBatch` method:
+```java
+     QueryOuterClass.Response response =
+                blockingStub.executeBatch(
+                        QueryOuterClass.Batch.newBuilder()
+                                .addQueries(QueryOuterClass.BatchQuery.newBuilder().setCql("INSERT INTO ks.test (k, v) VALUES ('a', 1)").build())
+                                .addQueries(
+                                        QueryOuterClass.BatchQuery.newBuilder().setCql("INSERT INTO ks.test (k, v) VALUES ('b', 2)").build())
+                                .build());
+
+```
+
+It takes the `Batch` as an argument. A Batch can contain N queries of type INSERT, UPDATE, or DELETE. We are adding two queries via `addQueries` method.
+See the [Batch Documentation] if you want to read more about how the Batch request is handled.
+
+### Async API
+
+Up to this point, we were using the blocking version of the generated stub. 
+We can also interact with the Stargate API using the async version of the stub.
+To do so, we need to pass the [StreamObserver] that will be called asynchronously when the results are available.
+
+Every StreamObserver needs to implement 3 methods: `onNext()`, `onError()` and `onComplete()`.
+For example:
+```java
+StreamObserver<QueryOuterClass.Response> streamObserver = new StreamObserver<QueryOuterClass.Response>() {
+           @Override
+           public void onNext(QueryOuterClass.Response response) {
+               try {
+                   System.out.println("response:" + response.getResultSet().getData().unpack(QueryOuterClass.ResultSet.class));
+               } catch (InvalidProtocolBufferException e) {
+                   throw new RuntimeException(e);
+               }
+           }
+           @Override
+           public void onError(Throwable throwable) {
+               System.out.println("Error: " + throwable);
+           }
+           @Override
+           public void onCompleted() {
+               // close resources, finish processing 
+               System.out.println("completed");
+           }
+       };
+```
+Please note that this is a very simplified version only for demonstration purposes and should not be used on production.
+
+Once we have the Observer, we can pass it to the `executeQuery` method on the async stub:
+```java
+stub.executeQuery(QueryOuterClass.Query.newBuilder().setCql("SELECT k, v FROM ks.test").build(), streamObserver);
+```
+This query will return immediately because it is non-blocking. 
+If your program (or test) is progressing to the end, you may not be able to see the results. 
+Your program may exit before the data arrives. 
+After some time, when the data arrives, the `streamObserver` will be called.
+
+The output of our program will look like this:
```

Comment:
Looks like we are calling `toString()` on opaque results: is this format documented/defined? It might also be good to know Java type of the result since that is needed to take any action (although for debugging String below is fine). From other documentation I think the default result type is CQL-specific.